app/assets/javascripts/Components/course_summaries_table.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import {render} from 'react-dom';
+
+import ReactTable from 'react-table';
+
+
+class CourseSummaryTable extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      grade_columns: [],
+    };
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+
+    $.ajax({
+      url: Routes.populate_course_summaries_path(),
+      dataType: 'json',
+    }).then(res => {
+      let grade_column = res.marks.map(c =>
+        Object.assign([],
+          c)
+      );
+      this.setState({data: JSON.parse(res.data),grade_columns:res.marks})
```

Comment:
We discussed in our meeting that you can remove the `JSON.parse`. Also the whitespace is wonky here.